release_notes/ocp_3_4_release_notes.adoc

```diff
@@ -42,43 +42,402 @@ To upgrade to this release from a previous version, see the xref:../install_conf
+based on available memory. With this release, eviction policy can now also be
+configured based on available disk.
+
+Nodes supports two file system partitions when detecting disk pressure:
+
+- The `nodefs` file system that the node uses for local disk volumes, daemon logs,
+etc. (for example, the file system that provides `/`)
+- The `imagefs` file system that the container runtime uses for storing images and
+individual container writable layers
+
+When configured, the node can report disk threshold violations, and the
+scheduler no longer tries to put pods on those nodes. The node ranks pods and
+then evicts pods to free up disk space.
+
+See
+xref:../admin_guide/out_of_resource_handling.adoc#admin-guide-handling-out-of-resource-errors[Handling Out of Resource Errors] for more details.
+
+[[ocp-34-storage]]
+=== Storage
+
+[[ocp-34-dynamic-storage-provisioning]]
+==== Dynamic Storage Provisioning Using Storage Classes
+
+Dynamic provisioning of persistent storage volumes for many storage providers
+was previously available in {product-title} as Technology Preview feature, but
+this release brings this feature into full support using the new _storage classes_
+implementation for the following:
+
+- OpenStack Cinder
+- AWS Elastic Block Store (EBS)
+- GCE Persistent Disk (gcePD)
+- GlusterFS
+- Ceph RBD
+
+See
+xref:../install_config/persistent_storage/dynamically_provisioning_pvs.adoc#install-config-persistent-storage-dynamically-provisioning-pvs[Dynamic Provisioning and Creating Storage Classes] for more details.
+
+[[ocp-34-security]]
+=== Security
+
+[[ocp-34-service-accounts-as-oauth-clients]]
+==== Service Accounts as OAuth Clients
+
+Users can now more easily integrate with the {product-title}-provided OAuth
+server from their own applications deployed within their project. You can now
+use service accounts as a scope-constrained OAuth client.
+
+See
+xref:../architecture/additional_concepts/authentication.adoc#service-accounts-as-oauth-clients[Service Accounts as OAuth Clients] for more details.
+
+[[ocp-34-scale]]
+=== Scale
+
+[[ocp-34-scalability-enhancements]]
+==== Scalability Enhancements for Logging and Metrics
+
+This release continues to add enhancements to drive improved scalability in
+{product-title} for larger cluster sizes, while still enabling users to leverage
+features like integrated, Hawkular-based metrics and EFK-based aggregate
+logging.
+
+- Current tested maximum ~= 12,000 pods / 220 nodes
+- Using the logging recommendations, you can run a 200 - 220 node cluster.
+- Using the metrics recommendations, you can run a 200 - 220 node cluster.
+
+[[ocp-34-networking]]
+=== Networking
+
+[[ocp-34-subdomain-wildcard-router]]
+==== Subdomain Wildcard Router
+
+Users can now use wildcard routes where all traffic destined to, for example,
+`*.foo.com` is routed to the same back-end service, configured to handle all the
+subdomains. You can specify that a route allows wildcard support through an
```

Comment:
+1. Rewritten to:

"Users can now use wildcard routes to determine the destination of all traffic
for a domain and its subdomains. For example, `*.foo.com` is routed to the same
back-end service, which is configured to handle all the subdomains."